src/jit/flowgraph.cpp

```diff
@@ -4761,19 +4761,37 @@ void Compiler::fgFindJumpTargets(const BYTE* codeAddr, IL_OFFSET codeSize, BYTE*
                 __fallthrough;
 
             case CEE_CALLI:
+
+                // CEE_CALLI should not be inlined if the call indirect target has a calling convention other than
+                // CORINFO_CALLCONV_DEFAULT. In the case where we have a no-marshal CALLI P/Invoke we end up calling
+                // the IL stub. We don't NGEN these stubs, so we'll have to JIT an IL stub for a trivial func.
+                // It's almost certainly a better choice to leave out the inline candidate so we can generate an inlined 
+                // call frame.
+
+                // Consider making this only for not force inline.
+                if (makeInlineObservations)
+                {
+                    CORINFO_SIG_INFO sig;
+                    info.compCompHnd->findSig(info.compScopeHnd, getU4LittleEndian(codeAddr),
+                                              impTokenLookupContextHandle, &sig);
+
+                    if (sig.getCallConv() != CORINFO_CALLCONV_DEFAULT)
```

Comment:
It likely doesn't matter, but `findSig` does a lot more work than just finding the calling convention. Would be good to verify that this is not a particularly hot path and the extra cost here doesn't show up in profiles.